CMakeLists.txt

```diff
@@ -0,0 +1,48 @@
+if(APPLE)
+    cmake_minimum_required(VERSION 3.5)
+else()
+    cmake_minimum_required(VERSION 3.2)
+endif()
+
+project(BoostMp11 VERSION 1.66.0 LANGUAGES CXX)
+
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+include(GNUInstallDirs)
+include(CMakePackageConfigHelpers)
+
+enable_testing()
+
+find_package(Boost 1.56 REQUIRED)  # 1.56+ for lightweight_test
```

Comment:
is this only required for tests?

if so, you can react on the user-controllable var `BUILD_TESTING[=ON]` which is set by `enable_testing()` and only depend on Boost when testing is requested.

Reply:
Mp11 depends on `boost/config.hpp` header so I can't cut dependency on Boost.

Reply:
I removed the dependency on Config by the way.

Reply:
I'll update the PR as @ax3l suggested

Reply:
@oliora do you want to update the PR description accordingly? :)